modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/ObjectRelationship/ObjectRelationship.js

```diff
@@ -12,44 +12,130 @@
  * details.
  */
 
-import Autocomplete from 'commerce-frontend-js/components/autocomplete/Autocomplete';
+import ClayAutocomplete from '@clayui/autocomplete';
+import {useResource} from '@clayui/data-provider';
+import ClayDropDown from '@clayui/drop-down';
+import {useDebounce} from '@clayui/shared';
 import {FieldBase} from 'dynamic-data-mapping-form-field-type/FieldBase/ReactFieldBase.es';
+import {fetch} from 'frontend-js-web';
 import React from 'react';
 
-const ObjectRelationship = ({
+const NETWORK_STATUS_LOADING = 1;
+const NETWORK_STATUS_UNUSED = 4;
+
+const LoadingWithDebounce = ({loading, render}) => {
+	const debouncedLoadingChange = useDebounce(loading, 500);
+
+	if (loading || debouncedLoadingChange) {
+		return (
+			<ClayDropDown.Item className="disabled">
+				{Liferay.Language.get('loading')}
+			</ClayDropDown.Item>
+		);
+	}
+
+	return render;
+};
+
+export function ObjectRelationship({
 	apiURL,
-	initialLabel,
-	initialValue,
+	initialLabel = '',
+	initialValue = '',
 	inputName,
-	itemsKey,
-	itemsLabel,
+	labelKey = 'label',
 	name,
-	onChange,
 	readOnly,
 	required,
-	value,
+	valueKey = 'value',
```

Comment:
These behaviors for the Field seem a little strange, normally the fields have a common contract.

- `value` will receive the current value
- `predefinedValue` is the initial value that is used when you don't have `value`
- `onChange` the callback will update the value in the Forms

There are other properties but they are not related to this case here.